prow/sidecar/censor.go

```diff
@@ -112,6 +116,218 @@ func (o Options) censor() error {
 	return kerrors.NewAggregate(errs)
 }
 
+// fileCensorer returns a closure over all of our synchronization for a clean handler signature
+func fileCensorer(sem *semaphore.Weighted, errors chan<- error, censorer secretutil.Censorer, bufferSize int) func(wg *sync.WaitGroup, file string) {
+	return func(wg *sync.WaitGroup, file string) {
+		wg.Add(1)
+		go func() {
+			if err := sem.Acquire(context.Background(), 1); err != nil {
```

Comment:
Not related to this PR but I forgot to mention that before: Instead of defaulting the concurrency to the number of cores, can we default it to some fixed const (4? 8?) so we do not have a different memory profile depending on how many cores the node we get scheduled to has?

Reply:
The pattern pretty much everywhere has benn to default to max cores, I don't really have a strong opinion other than it's likely better to be self-consistent with all the other places we do this same thing.

Reply:
Yeah but the reason for that is usually to limit file descriptors or such. Here, the reason is to have a limited and predictable memory profile. We need that to be able to set proper requests. We can not autochange requests based on the number of cores on the node a given pod gets scheduled to.

Reply:
Makes sense.